datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -611,6 +629,9 @@ impl Stream for GroupedHashAggregateStream {
                     match ready!(self.input.poll_next_unpin(cx)) {
                         Some(Ok(batch)) => {
                             let _timer = elapsed_compute.timer();
+                            if let Some(probe) = self.skip_aggregation_probe.as_mut() {
+                                probe.record_skipped(&batch);
```

Comment:
This is the actual call that records the metrics . The rest of the PR is comments and plumbing